File: polyglot/seis_native/nim/seis_native_kernel.nim
type
  NativeRoadmapItem* = object
    lane*: string
    score*: int
    language*: string

let roadmap* = @[
  NativeRoadmapItem(lane: "Apple First", score: 100, language: "Swift"),
  NativeRoadmapItem(lane: "Data AI", score: 88, language: "Python"),
  NativeRoadmapItem(lane: "Systems", score: 84, language: "Rust"),
  NativeRoadmapItem(lane: "Android", score: 76, language: "Kotlin"),
  NativeRoadmapItem(lane: "Windows", score: 72, language: "CSharp"),
  NativeRoadmapItem(lane: "Infrastructure", score: 70, language: "Go")
]

proc topLane*(): NativeRoadmapItem = roadmap[0]

File: polyglot/seis_native/fortran/seis_native_kernel.f90
module seis_native_kernel
  implicit none

  type :: native_roadmap_item
     character(len=32) :: lane
     integer :: score
     character(len=32) :: language
  end type native_roadmap_item

  type(native_roadmap_item), parameter :: roadmap(6) = [ &
     native_roadmap_item("Apple First", 100, "Swift"), &
     native_roadmap_item("Data AI", 88, "Python"), &
     native_roadmap_item("Systems", 84, "Rust"), &
     native_roadmap_item("Android", 76, "Kotlin"), &
     native_roadmap_item("Windows", 72, "CSharp"), &
     native_roadmap_item("Infrastructure", 70, "Go") &
  ]

contains
  function top_lane() result(item)
    type(native_roadmap_item) :: item
    item = roadmap(1)
  end function top_lane
end module seis_native_kernel
